### PR TITLE
feat(db): adds new init method for network

### PR DESCRIPTION
**Pull request by andrewxhill**

### CODE REVIEW

File: packages/database/src/db.spec.ts [hunk 1]
```diff
@@ -361,5 +362,31 @@ describe('Database', () => {
...
+      const store = new MemoryDatastore()
+
+      // We'll just use a dummy auth here. Hub auth should be tested in @textile/hub
+      const auth: UserAuth = {
+        key: '',
+        sig: '',
+        msg: '',
+        token: '',
+      }
+      const db = Database.withUserAuth(auth, store, {}, 'http://localhost:6007')
```
Comment (andrewxhill) [inline packages/database/src/db.spec.ts:376]: the withUserAuth default is remote. so here just using a local process for tests

Comment (andrewxhill) [reply packages/database/src/db.spec.ts:375]: given that these will update... should we start implementing a parallel method wherever this exists as, `refreshUserAuth(auth)`?

Comment (andrewxhill) [reply packages/database/src/db.spec.ts:375]: or maybe `updateUserAuth(auth)`

Comment (carsonfarmer) [reply packages/database/src/db.spec.ts:375]: We probably should yet, but it really doesn't feel right to have that sort of method on the database...

Comment (andrewxhill) [reply packages/database/src/db.spec.ts:375]: what are you thinking?

Comment (carsonfarmer) [reply packages/database/src/db.spec.ts:375]: I'm not sure yet :P We could certainly add it in for now... I'm just thinking through separation of concerns...

**Pull request merged**